pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -391,6 +409,8 @@ def default_compilation_pass(
         if optimisation_level == 0:
             if self._standard_gateset:
                 passlist.append(self.rebase_pass())
+            else:
```

Comment:
Probably we should test that the gateset is compatible here, rather than assume that it is if `self._standard_gateset` is `False` (and throw an error if not). Otherwise if a device comes along with a new gateset we'd blithely submit circuits with invalid gates.

I suggest a check similar to that for `_standard_gateset`. In fact, `_standard_gateset` is probably a bad name to use now. We could have an `Enum` of known gatesets instead (and store its value as a `Backend` attribute) instead of the current boolean.

Reply:
Good point, this should be done more carefully

I want to ensure that `IBMBackend.rebase_pass()` gives the right rebase too.